scalafix-core/shared/src/main/scala/scalafix/internal/util/PrettyType.scala

```diff
@@ -513,18 +500,17 @@ class PrettyType private (table: SymbolTable, shorten: QualifyStrategy) {
           Type.Tuple(targs)
         case _ =>
           val qual: Type.Ref = prefix match {
-            case Some(pre) =>
-              if (pre.tag.isSingletonType) {
-                Type.Select(toTermRef(pre), name)
-              } else {
-                Type.Project(toType(pre), name)
-              }
-            case _ =>
+            case s.NoType =>
               if (shorten.isName) {
                 name
               } else {
                 toTypeRef(info(symbol))
               }
+            case _: s.SingleType | _: s.ThisType | _: s.SuperType |
+                _: s.ConstantType =>
+              Type.Select(toTermRef(prefix), name)
```

Comment:
Can a constant type be a prefix to anything?

Reply:
I don't think so, removing it seems to be OK. This was a direct refactoring of the old implementation. Hurray for breaking down `SingletonType` 🎉